pilot/pkg/model/push_context_test.go

```diff
@@ -1972,6 +1973,103 @@ func TestServiceWithExportTo(t *testing.T) {
 	}
 }
 
+func TestGetHostsFromMeshConfig(t *testing.T) {
```

Comment:
can we add the exhaustiveness check we discussed?

Reply:
IMO, not.
once we add a test use reflect count there should have `N` ExtensionProviders, when we add a new `EntensionProvider` in the future, the `Automator: update istio/api@master` PR will be block.

Reply:
ok, I will trying to add a test.

Reply:
updated, please review again.